src/dotnet/HQ.Server/Services/HolidayServiceV1.cs

```diff
@@ -205,5 +212,62 @@ public async Task<Result<ImportHolidayV1.Response>> ImportHolidayV1(ImportHolida
             Updated = updated
         };
     }
+    public async Task BackgroundAutoGenerateHolidayTimeEntryV1(CancellationToken ct)
+    {
+        var today = DateOnly.FromDateTime(DateTime.UtcNow);
+        var holidays = _context.Holidays
+        .AsNoTracking()
+        .AsQueryable();
+        var jurisdicitons = Enum.GetValues(typeof(Jurisdiciton));
+        foreach (Jurisdiciton jurisdiciton in jurisdicitons)
+        {
+            // retrieve upcoming holiday based on today's date
+            var upcomingHoliday = await holidays.Where(t => t.Date >= today && t.Jurisdiciton == jurisdiciton).FirstOrDefaultAsync(ct);
+            if (upcomingHoliday == null)
+            {
+                return;
+            }
+            var staff = _context.Staff.
+            AsNoTracking()
+            .AsQueryable();
+            var vacationChargeCode = await _context.ChargeCodes.Where(t => t.Code == VacationChargeCode).FirstOrDefaultAsync(ct);
+            if (vacationChargeCode == null)
+            {
+                return;
+            }
```

Comment:
Move this outside of the for loop, no need to lookup for every jurisdiction. Also see note above about how to loo kit up, rename to holidayChargeCode.